internal/apis/certmanager/v1alpha2/types_certificate.go

```diff
@@ -225,6 +231,13 @@ type CertificateSpec struct {
 	AdditionalOutputFormats []CertificateAdditionalOutputFormat `json:"additionalOutputFormats,omitempty"`
 }
 
+// We currently support UPN or any String-like OID type using oid:x.x.x.x type and StringValue value.
+// You should ensure that the OID is valid for the string type as we do not validate this.
+type OtherNameSAN struct {
+	Type        string `json:"type,omitempty"` // can be "upn or oid:1.3.6.x.x"
```

Comment:
I'm not sure if we should support the name directly (eg, upn).
Is OID-only a good-enough solution?